tests/integration/targets/lookup_hashi_vault/lookup_hashi_vault/tasks/approle_test.yml

```diff
@@ -30,7 +30,7 @@
 
     - name: 'Failure expected when inexistent secret is read'
       vars:
-          secret_inexistent: "{{ lookup('community.general.:qhashi_vault', conn_params ~ 'secret=' ~ vault_kv2_path ~ '/secret4 auth_method=approle secret_id=' ~ secret_id ~ ' role_id=' ~ role_id) }}"
+          secret_inexistent: "{{ lookup('community.general.hashi_vault', conn_params ~ 'secret=' ~ vault_kv2_path ~ '/non_existent_secret4 auth_method=approle secret_id=' ~ secret_id ~ ' role_id=' ~ role_id) }}"
```

Comment:
Good spot, right now the failure happens because of the misspelling of the lookup name ;)

Reply:
One additional thing that caught me out was requiring a space on the end of `conn_params`. We should probably:
- put `conn_params` at the end or have a space before 'secret='

Reply:
Do you think adding a comment near where `conn_params` is defined pointing out the importance of the trailing space would be enough?

Reply:
Adding a comment in this PR is ok, but adjusting all use-cases of `conn_params` should be better done in another PR (if you want to do this).

Reply:
#1035  opened to address this